### DISCUSSION TITLE
[PLUGIN] To Do

### DISCUSSION THREAD

**Discussion by nicollasricas** [TriliumNext/Trilium#2946] [Show and tell]
# To Do ✔️

An improved version of the trilium default task manager ???

I've made this for myself and tough, why not share it? I consider this a `preview` version, so make sure you have your backups set.

Minimum required version v0.52.2

<https://user-images.githubusercontent.com/7860985/175755552-e633092f-661f-4907-8099-b99c5ff1c70d.mp4>

## Installation

- (Suggestion) Create a note called `Trilium` and inside another called `Plugins`.

- Install this [plugin](https://github.com/zadam/trilium/files/8983781/To.Do.-.1.0.0.zip) by right-clicking in the `Plugins` note, import into note, select the file to import, uncheck “Safe Import” and make sure “Read contents of .zip...” is checked.

- Install [Collection Views](https://github.com/mabeyj/trilium-collection-views) plugin (not required, but you are gonna miss a lot).

  - Add `#collectionViews` label to Collection Views root note.
    ![image](https://user-images.githubusercontent.com/7860985/175755718-2652effd-511e-4d43-953d-ba1e444669c7.png)

- Restart Trilium

**If you have any security/privacy concerns about this plugin, just open Trillium in safe-mode (blocks script execution) and look at what the code of this plugin does.**

## Getting Started

Create a note anywhere, add the `#initTodo` label, reload the front-end or restart Trilium.

- The note with the label will be the root of your to-do list.
- You can create multiple lists at the same time
- Completed tasks will be archived automatically after 7 days.
- The time spent tracking works based on minutes. If you work on a task for less than that, it won't track anything.

## To Do

      - To Do (root)
         - Archive
         - Backlog
         - Done
         - In Progress
         - Tags

- Archive - Completed tasks archived by year / month / day
- Backlog - Tasks on hold to be worked on
- Done - Nothing more to say, just good job!
- In Progress - Good session!
- Tags - It's tags, what else I could say? You could use this in every way you want. Separate tasks by priority (low and high)? As a separator for each app in a single to-do list? As status like bug, enhancement? The version you plan to release it? The sprint?
  - Tags can be customized, adding an attribute to the tag note.
    - `badgeBackground`
    - `badgeColor`

## Examples

Just a few examples on how you could use this.

![image](https://user-images.githubusercontent.com/7860985/175755829-58962138-a366-4815-a3cb-8f86c002fa22.png)

![image](https://user-images.githubusercontent.com/7860985/175755946-74cb8acb-3263-4988-9345-bb74d82a817b.png)

![image](https://user-images.githubusercontent.com/7860985/175756001-ff37ea34-674d-42c0-8dd2-832b7f5a4ae7.png)

`Inbox` - The Inbox note is not part of the plugin. It is just a child note that I manually create on each to do. I put anything related to the project that is not yet confirmed as an task. Could be ideas, examples, inspirations, sketches, toughs. I also don't care if stuff here is messier.

## Know Issues

- Changing attributes sometimes doesn't refresh the widget UI. I couldn't figure out a reliable way to identify attribute changing and refresh the widget. Changing the note's content or switching notes forces the refresh.
- Sometimes when a note is completed or archived, it appears as a cloned note, but the note was moved. Refreshing the front-end or restarting Trilium should solve, until I figure out a way to fix this.

That's it.

Good session!

Nicollas R.

[To Do - 1.0.0.zip](https://github.com/zadam/trilium/files/8983781/To.Do.-.1.0.0.zip)

**Comment by meichthys** [comment]
Nice work!
Related to #986

**Reply by 0xDTC** [reply]
But active task should be listed in check-list and should display on right side all the time there we can add task also just by clicking add button and once done we put mark on that task as i will save in previous task

**Reply by 0xDTC** [reply]
I would like it as a default feature in trilium this is way to help full

**Reply by nicollasricas** [reply]
> But active task should be listed in check-list and should display on right side all the time there we can add task also just by clicking add button and once done we put mark on that task as i will save in previous task

Sorry, but I didn't quite understand what you are asking. Which check-list? Display what on the right side? The button panel? If yes, then I did think about adding a button there, but since there could be multiple to-do lists how it's going to handle to which to-do list the task goes? I can't add a dropdown selection there.

**Reply by 0xDTC** [reply]
I can create DFD for this how todo checklist will work and sketch for how it will look on trilium

**Reply by nicollasricas** [reply]
> I can create DFD for this how todo checklist will work and sketch for how it will look on trilium

Ok, then let me know when it's ready. If somehow could improve the workflow that we already have, I'm up for it.

**Reply by 0xDTC** [reply]
> Ok, then let me know when it's ready. If somehow could improve the workflow that we already have, I'm up for it.

![image](https://user-images.githubusercontent.com/95960398/189092717-d5c001b4-f602-45fc-9f65-9bf052c90aa0.png)

**Reply by 0xDTC** [reply]
> Ok, then let me know when it's ready. If somehow could improve the workflow that we already have, I'm up for it.

for other this we can connect on discord if you are using. Discord id : Divine Clown#0029

**Comment by nicollasricas** [comment]
![Timeline](https://user-images.githubusercontent.com/7860985/179136147-5cd151de-2c2e-42d3-9560-3bc619cb315d.png)

> Please ignore the long named tasks, they are personal stuff that are protected by password and I didn't unlock the session (a feature that is not part of this plugin, it's built-in to Trilium).

# 1.1.0

## Changes

 - Added a timeline view
 - Fixed some bugs

## Know Issues

 - Completing tasks in the timeline make the left notes missing, just reload the frontend or switch notes. Everything is fine!!!

## Updating

- Every note scheduled will appear in the timeline, it doesn't matter from which project or where it's located. To schedule a note, just change the `due date` attribute on the note.

- Before updating, delete the older version of the plugin. You will be warned about relations been broken on deleting the plugin, don't worry, there's a migration script that will fix the relations upon updating.

- Update the plugin as if you were installing for the first time.

- To create a timeline, just create a note and add the `#initTimeline` label and refresh the front-end.

[To Do - 1.1.0.zip](https://github.com/zadam/trilium/files/9117544/To.Do.-.1.1.0.zip)

**Reply by Alumniminium** [reply]
super sweet, thanks

**Reply by jaduff** [reply]
Does this still work with version 0.6+? I've installed the plugins, restarted, and added the '#initTodo' tag and refreshed the interface, and even restarted. Nothing happens, no alert saying it succeeded, and no sub-note generation.

**Reply by jaduff** [reply]
Looks like there has been an api change, runonServer has been replaced with api.runonBackend.

**Comment by SergioRibera** [comment]
Maybe could suggest using events to know when the `#initTodo` label is added to a note, so that there is no need to reload the interface or restart the application or even on right-click as a new element

**Reply by nicollasricas** [reply]
I don't think this could be achieved using events. The events are targeted or child notes related, unless I add the event attribute to trilium root note and monitor changes from there. I don't know if there's any downside of doing that. @zadam ?

**Reply by zadam** [reply]
I'm not sure what exactly you need, but you could take a look at this: https://github.com/zadam/trilium/blob/master/src/public/app/widgets/buttons/edit_button.js#L41-L49

`entitiesReloadedEvent` is not documented in widget API, but it's very widely used event handler which allows you to listen to all entity changes.

**Comment by TheBig-O** [comment]
Hey, I'm really liking the idea behind this ToDo list, but I'm struggling to figure out how to get started after the installation and basic structure is set up. I created a note with #initTodo label and reloaded the frontend. 
At that point, I have the Archive, Backlog, Done, In Progress, Tags notes. What I don't understand how to do is create a Task that works within this structure. I tried creating an Inbox note with an inherited template attribute that points to the "task" note, but that didn't work. 
If you could give me a few tips on what I'm missing, that would really help. I'm sure it's something simple!
Thanks for the great work on this!

**Reply by 0xDTC** [reply]
Create some special combination or mark that will be recognised as todo mark

**Reply by TheBig-O** [reply]
I'm not sure what you're telling me? I'm trying to figure out where or how to create task in this folder/note structure so that as it progresses it tracks the time, moves from "in progress" to "Done" and on to "Archive."
Since each of those are render notes, there must be some way that they know when something has changed. That's the part I'm trying to figure out. Where do I start a new now so that the plugin knows I've created a task?
It looks like if I create a note directly under "in progress," things start happening. Is that the starting point? What if I have old tasks I want to add to the "Done" section? Will that still work?
Like I said, I'm sure it's something simple that I'm missing.

**Reply by 0xDTC** [reply]
Then assign ids to every todo task for tracking

**Reply by TheBig-O** [reply]
Are these IDs that I just make up myself or does the plugin do it? Will they show up in the overall listing as "Done" or are they just placeholders? 
As an example, if I copied all the completed task from the base Trilium app into the "Done" folder, will those be shown in the table or do I have to do something additional to make that happen?
I appreciate your help figuring this out. I'll just try a few things and see what happens.

**Reply by nicollasricas** [reply]
Hey!

Just create the to-dos inside any of the categories that are created. The 'Inbox' is just a suggestion I've made, and also uses to have notes and thoughts that are not already assigned to a specific task, like a simple notepad.

This may help:
<https://user-images.githubusercontent.com/7860985/175755552-e633092f-661f-4907-8099-b99c5ff1c70d.mp4>

**Reply by nicollasricas** [reply]
You add the tasks you are going to do to the backlog if you aren't familiar with that structure or haven't been using kanban or similar. Everything else is contextual and done automatically based on progression. Just use the features available by this plugin.

- Backlog is your starting point.
- Tasks added to done section are set as done automatically on the creation date.
- Tasks added to in progress starts tracking immediately.
- Tasks added to archive are archived automatically if their done date is set.
- Done tasks are archived automatically after 7 days.

You don't have to worry about adding specific attributes or anything like that to notes, except for the initials:

#initTodo → Create the to-do itself
#initTimeline → Create a timeline visualizer, not required.
#badgeColor, #badgeBackground → for customizing tags, not required.

**Reply by TheBig-O** [reply]
Thank you for that. I had sorted out some of what you wrote by trial and error but your clarifications really helped. 
This is a great addition for tasks!  I appreciate your help and the excellent plug-in!

**Reply by TheBig-O** [reply]
Good Morning, would you be able to give an example of how to use the "Tags" section? I've added various tags to my ToDos but nothing changes in that section. (It just shows a blank page.) Specifically, what does "#badgeColor, #badgeBackground → for customizing tags, not required" do for my todo items?
Do I need add something specific to the Tags note? 
Sorry for the basic questions. Just not a Javascript person and don't know what's supposed to happen.
Thank you!

**Reply by 0xDTC** [reply]
Bro use some js library which can create unique id and assign to the todo task once task is finished it will mark done all over the where ever that todo task existed.

**Reply by TheBig-O** [reply]
I'm not sure what this means? How would a unique ID affect the tags page?
As it is, when I add a date to the promoted date completed input, it marks the task as complete. My question was related to how to interact with the "Tags" page and why it isn't showing anything.

**Reply by nicollasricas** [reply]
> I'm not sure what this means? How would a unique ID affect the tags page? As it is, when I add a date to the promoted date completed input, it marks the task as complete. My question was related to how to interact with the "Tags" page and why it isn't showing anything.

Hey,

Just create a child note inside the Tags section. Each note is a tag that can be used on the tasks. The attributes #badgeColor and #badgeBackground allow you to customize the color, that's why I've said that they are not required.

Here's how the tags should look and how you use them:

![image](https://user-images.githubusercontent.com/7860985/220689649-6d70fe96-0ccc-46b8-9b69-121a8ddd5856.png)

You can look at this video (**0:50**), I show how to use tags and also how to customize their background and foreground colors.

**VIDEO:**
<https://user-images.githubusercontent.com/7860985/175755552-e633092f-661f-4907-8099-b99c5ff1c70d.mp4>

**Reply by TheBig-O** [reply]
Thank you! This is exactly what I was looking for. I was thinking the tag notes would auto create as in the built-in Task Manager. This format gives great flexibility!
I appreciate your help... again.

**Reply by TheBig-O** [reply]
For some reason the process you describe and show in the video isn't working. I created a child note and named "Rumplestilskin" (wanted a new tag I hadn't used before.) After doing that, I assigned both a badge color and background to the tag note using lower case color names. (standard blue and white) I then assigned Rumpelstiltskin to the task and refreshed the front-end. 
After the reload, all the tags are still the same basic gray color.
Is there a step I'm missing? I'm using the Mac and Web versions. Neither shows a change in tag color.

**Reply by nicollasricas** [reply]
![image](https://user-images.githubusercontent.com/7860985/220744420-496bee36-a595-40d2-9a81-17b93cdedaa9.png)

It should look like this. There isn't a need to reload after adding tags. If still not working, try to use hex colors instead.

Hex colors example:

`#badgeColor="#fff" #badgeBackground="#000"`

_Don't forget the double quotes._

**Reply by TheBig-O** [reply]
Not sure what's going on. I went back through to make sure there wasn't anything getting in the way, so to speak and it looks like there's nothing that might interfere in the tags or the notes themselves. Here's a couple of screenshots to show how I have it set up.
Thanks for helping me figure out what's going on.
![Tag_Properties](https://user-images.githubusercontent.com/25670035/220755865-e20517f3-b4ac-4528-9cac-580f38d650f1.jpg)
![Progress-view](https://user-images.githubusercontent.com/25670035/220755861-99bff0a4-6a7c-40fc-9e77-ab34dd74f293.jpg)
![Task_Properties](https://user-images.githubusercontent.com/25670035/220755868-2936c9f6-7ec3-4246-8bd3-7da16aebfac0.jpg)

**Comment by nicollasricas** [comment]
Did you add the tag to the note manually? If so, I would not recommend, since everything is done automatically to ensure all relationships are in place.

The tag must be a relationship and not an attribute as you have set.

![image](https://user-images.githubusercontent.com/7860985/220781892-38bff06b-6b25-4841-972f-ccb69ee482da.png)

`#tag=Tag1` An attribute with name tag and value Tag1.
`~tag=Tag1` A relationship with name tag and value Tag1 **(MUST BE A NOTE)**.

This is how it should look like:
![image](https://user-images.githubusercontent.com/7860985/220782648-2e86023c-f359-4c96-bd0c-3770490f769d.png)

**Reply by TheBig-O** [reply]
That was it! 
I saw the buttons at the buttons at the bottom of the center pane in your video but hadn't seen them on my own screen. As soon as I clicked that, I saw a pop up of the various tags I had created. Clicked one and it worked perfectly. (I had been creating the tags manually.)
Sorry for the confusion but thank you for the help!!

**Reply by nicollasricas** [reply]
Glad it worked.

As I've mentioned in another comment [here](https://github.com/zadam/trilium/discussions/2946#discussioncomment-4970623), you don't need to do anything manually except for the initial attributes to create the list itself, everything else is taken care of.

If the bottom bar (widget bar) doesn't show, the task was created manually without the right attributes and relationships.

**Make sure to remove the tag attributes you have added to those tasks.**

**Reply by TheBig-O** [reply]
>  "If the bottom bar (widget bar) doesn't show, the task was created manually without the right attributes and relationships."

... This was the part I missed. 
Thanks again for your help!

**Comment by Joseto58** [comment]
Hello! Thank you for this amazing Plugin. I added a small script so that tasks may generate labels to be used by collection-views to render text and I wanted to share it:

One of the ways in which I'm using this plugin is by creating a table with all the components of a project, along with the area they belong to(using tags) and their completion status.

I created a column with a short description for each task on the table, which helped a lot when discussing planning with my clients.

I wanted to automate the process of generating this table with the collection-views table, however, I was struggling to find a way to add the tasks description into the table, and since the table is rendered you can not add text manually. From what I gathered the solution was using attributes to store the descriptions of the tasks, which was very messy using promoted attributes and time consuming.

---

### Here's what I came up with:

Tasks now have a promoted boolean called "addDescription" that generates labels from the notes content:
![image](https://user-images.githubusercontent.com/122640750/236942283-7b4dcc90-66c2-4afd-9747-bde0d59a2648.png)

The created label "descriptionLabel" is a multi text label, which can be included in collection-views renders:
![image](https://user-images.githubusercontent.com/122640750/236944136-45709e7d-e3da-4e36-8715-12d3b2b06084.png)

---
### How to use:

Descriptions are generated from the first lines of text of a note, you must add a line "---" to indicate where the description ends, as seen on the picture above.
The text that is inside this area will be separated by paragraphs (< p > aka. pressing enter into a new sentence), you can have one block of text, or multiple lines.
Disabling the bool will remove the labels from the note, and activating it will delete all existing "descriptionLabel" entries before generating new ones.

---
### Set-up

Add the following promoted attribute to the task template note (Plugins/To do/Templates/Task):
"#label:addDescription=promoted,single,boolean"

![image](https://user-images.githubusercontent.com/122640750/236933149-0facba81-5343-4f3a-9a16-c2509f3ea75e.png)

Add the following code block to the ***attributeChanged*** script (Plugins/To do/Scripts/attributeChanged):

```javascript
  if (attributeName === "addDescription") {
    if(api.originEntity.value === "true"){
      await descriptionLabel(taskNote.noteId);
        } else {
      taskNote.removeLabel("descriptionLabel"); 
        }

    return;
  }    
```

And add a child note to it called called "descriptionLabel" of type JS backend with the following script:

```javascript
module.exports = async function (taskNoteId) {
  const taskNote = await api.getNote(taskNoteId);

  if (!taskNote) {
    return;
  }


//const taskNote = api.originEntity;
const content = taskNote.getContent();
var text = String(content);
var parts = text.split(/<hr[^>]*>/i);


text = parts[0];

text = text.replaceAll("</p>","");
text = text.replaceAll("&nbsp;"," ");    
parts = text.split("<p>");

//Should check if there is a description at all, otherwise, return... or erase?

//Array of description points
parts.shift();
const size = parts.length;
const myArray = new Array(size);

for (let i = 0; i < size; i++) {
    myArray[i] = parts[i];
}

//Remove all pre-existing values of the label before adding the updated ones.
  taskNote.removeLabel("descriptionLabel"); 

for (let i = 0; i < size; i++) {
  taskNote.addLabel("descriptionLabel", myArray[i]);
}

};
```


---

### Disclaimer

I'm sharing this in case someone finds it useful or could make it better/more intuitive. The format that the description must have is janky and there is no safeguard in case the script gets called and there is no description or line to stop the script which could result in unexpected behaviour.  
I'd love to see something like this integrated in a cleaner way, but this approach works wonderfully for my use case. Cheers!

**Comment by bartv2** [comment]
Is there a script to migrate the demo task manager tasks to this? Or are there easy steps to follow?

**Comment by spirit1431007** [comment]
Nice work! I just replace all `runOnServer` to `runOnBackend` and remove `api.dayjs.duration`
[To.Do.-.1.1.0.zip](https://github.com/zadam/trilium/files/12486290/To.Do.-.1.1.0.zip)

**Reply by renyuneyun** [reply]
Hi. Thanks for sharing. However, do you have any ideas to deal with the error about `async` and `api.runOnBackend()`? (See [this comment below by bonnepioche](https://github.com/zadam/trilium/discussions/2946#discussioncomment-8119061))

**Comment by adoreparler** [comment]
Working on something personal that is a widget for this plugin if anyone is interested.
https://github.com/zadam/trilium/discussions/4289

**Comment by bonnepioche** [comment]
Hi,
I have installed the version 1.1.0 and I have the following error message now when I launch Trilium:
![image](https://github.com/zadam/trilium/assets/16622600/0dbbde94-ad37-40f8-8d28-f6a28d65943e)

**Reply by renyuneyun** [reply]
Same issue here. I tried to change that to `api.runAsyncOnBackendWithManualTransactionHandling()` ([doc](https://zadam.github.io/trilium/frontend_api/FrontendScriptApi.html)), and noticed `api.transactional()` ([doc](https://zadam.github.io/trilium/backend_api/BackendScriptApi.html)). However, it looks like (based on doc) `api.transactional()` requires synchronous function as well, so I'm not sure if wrapping async function there is a good idea?

**Reply by banjuer** [reply]
mark
same issue

**Reply by StridentDefender32** [reply]
[To.Do.-.1.1.0-fixedBackendCalls.zip](https://github.com/zadam/trilium/files/15194990/To.Do.-.1.1.0-fixedBackendCalls.zip)

I also ran into this issue and took a stab at fixing it myself. I've not tested this very thoroughly, but it at least appears to work.

Also, maybe I missed something, but I did not see anything that made me believe `api.transactional()` needed synchronous functions. I did not attempt to use synchronous functions and just wrapped what appeared to be the necessary parts.

**Reply by renyuneyun** [reply]
@StridentDefender32 I think the reason for `api.transactional()` requiring synchronous function is within its name ([database transaction](https://en.wikipedia.org/wiki/Database_transaction)). It would, probably, call the function (the parameter) assuming it is a transaction -- an action that has its own complete meaning and purposes, and is not related any other actions. Probably trilium uses this to separate the states of multiple concurrent calls, for consistency.

**Comment by feelancer21** [comment]
@nicollasricas is it possible to add more sprint columns by my own? Like to have something like a product backlog and a sprint backlog.

**Reply by nicollasricas** [reply]
It should be easy to-do it if you know how to use the collection-views plugin, which is used by this one I've made. Just read their documentation, it's pretty straightforward. Everything is made by Trilium attributes/relations system and the scripts are just to automate stuff.

I'm no longer working on this plugin since I've moved to Obsidian, I did have a huge updated in works, but it's abandoned and not have any plans to finish it. I can't find it anymore anyway, so also can't upload here for someone to take the implementation.

**Reply by feelancer21** [reply]
Thanks for the hint. Will check it out.

**Reply by CobriMediaJulien** [reply]
@nicollasricas Small question: Why did you switch :)?

**Reply by adoreparler** [reply]
I switched also to Obsidian for mobile app

**Reply by meichthys** [reply]
> I switched also to Obsidian for mobile app

Thanks for the feedback 👍

**Reply by TheBig-O** [reply]
When you switched, how did you do it? Were you able to get all of your notes carried over in a useful manner? 
Not sure I want to switch but would like to know how to do it so I can test obsidian a bit more!
Thanks!

**Reply by zerebos** [reply]
Hopefully TriliumNext will be able to release a mobile app as well.

**Reply by nicollasricas** [reply]
@CobriMediaJulien @TheBig-O @meichthys 

Trilium is an amazing tool and served me well for a long time, but it lacks extensibility. While it does support scripts and certain levels of customization, the way it handles these features didn't quite work for me. Some people have managed to create plugins outside the Trilium "notes scope", but the whole process of developing plugins feels awkward. That was the first reason for my decision. The second reason was that after Zadam decided to stop working on the app, I felt uncertain about its future. Although I haven't encountered any significant bugs in Trilium. It's quite stable! There were a few issues regarding improving its extensibility. Unfortunately, none of these issues were addressed in production, and with development halted, they probably never will be.

I didn't try to automate note exporting/importing or anything like that. I did everything manually and used this migration as an opportunity to clean up some unused content.

If you decide to install Obsidian, I recommend making sure it fits your needs before fully switching over. If something is missing by default, search for a plugin that can meet your requirements. Create a few test notes to try it out before fully migrating.

Obsidian is not open-source, but it has a large community that regularly supports and adds features through its extensions SDK. It's simpler to use since it doesn't require databases, all notes are plain markdown files. There are many cool plugins and themes available, and using these plugins, I was able to control everything I needed (syncing, backup, encryption) and The best part is that it has a native mobile application.

If, for any reason, Obsidian changes in a way that doesn't fit my needs anymore, I have complete control over the syncing, encryption, and backup processes, so I could either migrate to another platform or continue using older versions.

The only thing I missed in Obsidian was encryption when handling my own syncing. To address this, I created a plugin for myself that keeps notes encrypted on the hard drive at all times. They are decrypted on the fly as I use Obsidian, and this solution also supports mobile. While there are a few plugins that encrypt your notes during syncing, they leave the notes unencrypted on your hard drive. The ones that can keep them encrypted on the hard drive use GPG, which doesn't work on mobile. Obsidian's cloud sync does support end-to-end encryption, which is useful for those who plan to use it.

That's it.

Good session to you all.

**Reply by meichthys** [reply]
> There were a few issues regarding improving its extensibility. Unfortunately, none of these issues were addressed in production, and with development halted, they probably never will be.

As per our first TriliumNext community meeting, this is probably the #1 priority of TriliumNext after the code is converted to typescript and development pipelines are ironed out, so this should improve relatively soon.

**Comment by deteeegit** [comment]
Thank you for this useful tool! 
I'm having an issue applying a tag to a task. I have the tag (called "consulting" no quotes) created in the Tags object. I have an in-progress task called "Diagram Process". I have a due date set. I go to apply a tag using the status/tags toolbar at the bottom, hit the tag icon, and ... nothing happens? It doesn't bring a popup panel up of the created tag.
EDIT: I should add that I followed each step in your demo video above and still nothing happens when I click on the tags button. No panel showing tags I added pops up. The Archive button seems to work. The Complete button seems to work. The time recorder works. The tag button does not, and the three-dot icon also does nothing for me...

**Reply by TheBig-O** [reply]
I'm in the same boat. I think something changed in the most recent releases. I've seen several oddities with the latest TriliumNext releases. In the web app, I no longer see tabs in Edge explorer. As you point out, the Tags menu does not work and I cannot see any of the other menu items at the bottom either.

**Reply by TheBig-O** [reply]
@nicollasricas, can you give us a pointer as to why the TAGS menu is no longer popping up? The JS and CSS define the dropdown in the task page, but I'm not seeing where JQuery or Bootstrap are linked to them? I looked through it but not sure how you linked into Trilium's built-in capabilities.
Thanks for putting this together. Hopefully, we can get it running again.

**Comment by liuxiawei** [comment]
It's a great job. But I cannot found version for TriliumNext? Is there anyone can help?

**Comment by TheBig-O** [comment]
I updated the code on version 1.1 of this plugin so that it works for me. I'm not promising anything for others who try my version, but I'm willing to help if I can.
This version fixes the popup menu showing to select tags for your tasks.

- Dropdown button didn’t have a working click handler for the menu.
```
// --- Dropdown toggle ---
this.$widget.find('.dropdown.note-actions button').click(function (e) {
  e.stopPropagation();
  $(this).next('.dropdown-menu').toggleClass('show');
});

// Close dropdown when clicking outside
$(document).click(() => {
  $('.dropdown-menu').removeClass('show');
});
```  
There are other minor fixes in the code, but the menu was the key concern.
I increased the version number to ensure people know they've got a revised version
[To Do 1.2.zip](https://github.com/user-attachments/files/22703436/To.Do.1.2.zip)

**Reply by TheBig-O** [reply]
Okay, my limited Javascript skills created an issue when I "fixed" the pop-up list. All other dropdown menus in Trilium stopped working.  So, I went back and added some isolation to the script that should eliminate the issue.  
If you see any problems, let me know and I'll take another swing at it.
I incremented the script to distinguish it, but really I only changed the file named `task` at this location: `To Do 1.3 / Widgets / task`.  
If you're looking for the shortcut, just swap out the code in that file.  
[To Do 1.3.zip](https://github.com/user-attachments/files/22830810/To.Do.1.3.zip)

**Reply by banjuer** [reply]
It seems to have expired on version 0.101.1

**Reply by TheBig-O** [reply]
I used it earlier this evening with the Kate's version and it seems to be working still. 
I just double checked and it's still working as expected. 
Make sure you're using version 1.3

**Comment by juliaroquette** [comment]
Has anyone made this work for Trilium Next? I managed to make the base TODO work, but Collections is now a part of Trilium Next (so Collections View is no longer a thing) and this seems to have broken the visualisation of the Tasks, etc.

**Reply by TheBig-O** [reply]
If you see my last entry, you'll find an updated version that works for me.(I'm running the latest release.)    
The collections view you're referencing is different from the view used in this plugin. They are not related.

**Reply by juliaroquette** [reply]
So this means I still need to add the Collection View plugin as described?

**Reply by TheBig-O** [reply]
Yes, the original installation instructions still apply.
